src/transformers/utils/hub.py

```diff
@@ -222,7 +222,7 @@ def extract_commit_hash(resolved_file: Optional[str], commit_hash: Optional[str]
     """
     if resolved_file is None or commit_hash is not None:
         return commit_hash
-
+    resolved_file = str(Path(resolved_file).resolve().as_posix())
```

Comment:
```suggestion
    resolved_file = str(Path(resolved_file).as_posix())
```
The `.resolve()` call will change a synmlink to the file it links to, and the whole cache system is based on symlinks. So this is what is breaking the tests.